Speed up `Line2D._edit_get_rect()`

arkology opened the pull request:
This method is called constantly on viewport movement and when `Line2D` node is selected.
Current implementation is not bottleneck for editor, but objectively not as effective as it could be even without doing over-optimization.
This PR replaces excessive `expand_to()` calls in for-loop with finding minimum and maximum positions among all points and then creating `Rect2D`. 

For one `Line2D` node with 116 points, single `Line2D._edit_get_rect()` call.

| Editor build | Before, average time, ns | After, average time, ns | Speed gain |
| --- | --- | --- | --- |
| Dev build | 9048 | 4756  | ~x1.9 faster |
| Production build | 2405 | 1150 | ~x2.1 faster |

arkology committed:
Speed up `Line2D._edit_get_rect()`

Ivorforce commented:
> Results are for dev build, so difference in optimized build may be not so big, but still should be.

I definitely recommend running the test on a non-dev build, because non-optimized builds are rarely representative of optimized ones. 

`expand_to ` is implemented in `rect.h` and thus fully visible to the compiler. Since it's dealing with floats, it's unlikely the call will be perfectly optimized - but it's good practice to check anyway :)

KoBeWi commented:
Related: #100418

arkology commented:
> I definitely recommend running the test on a non-dev build, because non-optimized builds are rarely representative of optimized ones.

Sure! Added results to OP. 

> `expand_to ` is implemented in `rect.h` and thus fully visible to the compiler. _Theoretically_, that means it could be optimized away - it probably isn't because it's dealing with floats - but it's good practice to check anyway :)

`min()`/`max()` are also implemented in header so I expected to get comparable results (and the are comparable).

groud reviewed (approved):
The optimization looks good for now.

Seeing @KoBeWi 's comment, I think we should implement a few things to improve the state of the editor:
- If we keep the rect, it should likely edit the points instead of the scale.
- If we remove the rect, it would be nice to have multi-points select.

In any case, we would still likely need a way to move the "origin" of the points, as I believe it's a valid use case.

akien-mga commented:
Thanks!
